src/handlers/authentication.js

```diff
@@ -0,0 +1,41 @@
+import Boomify from 'boom';
+import Boom from 'boom';
+
+const functions = Object.create({});
+
+functions.authenticateEmail = async (request, h) => {
+
+    const server = require('../server.js');
+
+    const emailToAuthenticate = request.params.email;
+
+    try {
+        // Get all users from Firebase
+        const userRefs = await server.db.collection('users').get();
+
+        // Keep id and user data
+        const users = userRefs.docs.map((user) => Object.assign({ id: user.id }, user.data()));
+
+        let authorizedUserId = null;
+
+        // Check if any user emails match given email
+        for (let i = 0; i < users.length; ++i) {
+
+            if (users[i].email === emailToAuthenticate && users[i].isActive === true) {
+                authorizedUserId = users[i].id;
+                break;
```

Comment:
You can probably return in here. Save some clock cycles. Pretty minor

return h.response(authorizedUserId); in for loop

Reply:
Nah I tried and for some reason it wouldn't return here.